src/backend/upload_file/upload_file_for_chunks.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import json
+import os
+from minio import Minio
+from minio.error import S3Error
+from fastapi import HTTPException
+import PyPDF2
+import pdfplumber
+from docx import Document
+
+# Logger setup
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+class DocumentReader:
```

Comment:
For all the important formats like pdfs, word documents, ppts, html files and other important file format, use separate document readers

For rarely used formats, try to check whether docling supports the reading of the files and providing the text output in our required format, if yes, we can use one docling library to read all other file types. 

Try to write the document reader for the below files and test it

If a filetype  text retrieval is not possible, lets see if its an important one, we need to implement or we can deprirotize that file type

1. Text-Based Filetypes
Simple and Structured Text Files
These are foundational formats for natural language processing tasks.

.txt: Plain text files, ideal for line-by-line or paragraph-based chunking.
.csv: Useful for tabular data, where rows or columns can be chunked.
.tsv: Tab-separated values; similar use to CSV but with tab delimiters.
.json: Hierarchical and structured data; suitable for nested embeddings.
.xml: Metadata-heavy or hierarchical content; requires parsing and cleaning.
.html: Web content; processed with HTML parsers to extract main text.
Advanced Text Formats

.md: Markdown files for knowledge bases or documentation.
.yaml: Configuration or structured data files.
.log: Log files for chunking events, errors, or sequential data.
Use Cases:

Knowledge bases, FAQs, search engines, chatbot training datasets.
2. Document Filetypes
Text-Rich Documents
Great for indexing large, complex documents.

.pdf: Scanned and digital documents; OCR may be required.
.docx / .doc: Word processor files with rich formatting.
.odt: OpenDocument text files.
.rtf: Rich text format files, widely used in legacy systems.
Presentations and E-Books

.pptx / .ppt: Slide presentations; chunk by slide or section.
.key: Keynote presentations; similar chunking as PPT files.
.epub: E-books; process by chapter or section.
.mobi: Amazon Kindle e-book format.
Specialized Document Formats

.tex: LaTeX files, typically scientific or academic papers.
.chm: Compiled HTML help files for knowledge retrieval.
Use Cases:

Enterprise document search, e-learning, scientific paper indexing.
3. Image Filetypes
Raster Images
Images requiring OCR or embeddings for similarity tasks.

.jpg, .jpeg, .png, .bmp, .gif: Standard image formats.
.tiff: High-quality images, often used in documents.
.ppm, .pgm: Portable image formats for research or niche use.
Vector Images

.svg: Scalable vector graphics with metadata and annotations.
.ai: Adobe Illustrator files; complex vector images for design retrieval.
Use Cases:

Document scanning, logo search, design inspiration databases.
4
Podcast search, call transcription, audio similarity systems.




Use Cases:

Educational content, video recommendation systems, caption-based search.
6. Code and Configuration Filetypes - A normal text reader should be good I think for these file types
Programming Languages
Chunk by classes, functions, or modules.

.py, .java, .cpp, .js, .ts: Common code formats.
.php, .rb, .cs: Web and application programming languages.
.sql: Database query files; chunked by queries or statements.
Configuration and Documentation

.yaml, .json, .toml, .ini: For storing configuration data.
.ipynb: Jupyter notebooks, chunked by cells.
Use Cases:

Code search engines, debugging tools, developer assistant systems.
7. Specialized Formats
Spreadsheets
Chunked by rows, columns, or sheets.

.xls, .xlsx: Microsoft Excel files.
.ods: OpenDocument spreadsheets.
.csv: Flat spreadsheet data.
Scientific and Mathematical Data